qiskit_experiments/database_service/utils.py

```diff
@@ -74,6 +75,47 @@ def utc_to_local(utc_dt: datetime) -> datetime:
     return local_dt
 
 
+def objs_to_zip(filenames: List[str], contents: List[str]) -> bytes:
+    """Write a list of files and contents to a zipped file.
+
+    Args:
+        filenames: List of names for the files to be zipped.
+        contents: List of file contents to be zipped in string form.
+
+    Returns:
+        A bytes object containing the zipped files.
+    """
+    zip_buffer = io.BytesIO()
+
+    with zipfile.ZipFile(zip_buffer, "w", zipfile.ZIP_DEFLATED, False) as zip_file:
+        for file_name, data in zip(filenames, contents):
+            zip_file.writestr(file_name, data)
```

Comment:
Thanks for suggesting this. I tried adding `zlib.compress(zip_buffer.read())` to the end of the method. For a 27 Q parallel T1 experiment, it only reduced `curve_data.zip` from 124 KB to 122 KB, and since this is more inconvenient for the user to decompress directly from the IQP interface, I'm not sure it's worth it. I also tried compressing each JSON file individually but the performance was even worse.